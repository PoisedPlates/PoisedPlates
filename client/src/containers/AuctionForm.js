import React, { Component } from 'react';
import { connect } from 'react-redux';

/* * Utils * */
import axios from 'axios';
import { Link } from 'react-router-dom';
import { Field, reduxForm, formValueSelector } from 'redux-form';

/* * Actions * */
import { postAuction, selectImage, deselectImage } from '../actions';

/* * Styles * */
import { Card, CardMedia } from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';
import ImagePhotoCamera from 'material-ui/svg-icons/image/photo-camera';
import MenuItem from 'material-ui/MenuItem';
import { AutoComplete as MUIAutoComplete } from 'material-ui';
import {
  AutoComplete,
  DatePicker,
  TimePicker,
  SelectField,
  TextField
} from 'redux-form-material-ui';

const states = [
  'AL','AK','AZ','AR','CA','CO','CT','DC','DE','FL',
  'GA','HI','ID','IL','IN','IA','KS','KY','LA','ME',
  'MD','MA','MI','MN','MS','MO','MT','NE','NV','NH',
  'NJ','NM','NY','NC','ND','OH','OK','OR','PA','RI',
  'SC','SD','TN','TX','UT','VT','VA','WA','WV','WI','WY'
];

const categories = [
  ['antiques', 'Antiques'],
  ['appliances', 'Appliances'],
  ['arts+crafts', 'Arts & Crafts'],
  ['atv/utv/sno', 'ATVs, UTVs, & Snowmobiles'],
  ['auto parts', 'Auto Parts'],
  ['baby+kid', 'Baby & Kid'],
  ['beauty+hlth', 'Beauty & Health'],
  ['bikes', 'Bikes'],
  ['boats', 'Boats'],
  ['books', 'Books'],
  ['cars+trucks', 'Cars & Trucks'],
  ['cell phones', 'Cell Phones'],
  ['clothes', 'Clothes'],
  ['computers', 'Computers'],
  ['electronics', 'Electronics'],
  ['farm+garden', 'Farm & Garden'],
  ['furniture', 'Furniture'],
  ['general', 'Miscellaneous'],
  ['household', 'Household'],
  ['camping', 'Camping'],
  ['tools', 'Tools'],
  ['toys+games', 'Toys & Games']
];

const styles = {
  outer: {
    padding: '5rem'
  },
  inner: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 'auto',
    width: '100%',
    height: '100%'
  },
  align: {
    verticalAlign: 'bottom'
  },
  button: {
    margin: 8
  },
  imageInput: {
    cursor: 'pointer',
    position: 'absolute',
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    width: '100%',
    opacity: 0,
  },
  card: {
    width: '80%'
  },
  state: {
    width: 80
  }
}

// validation functions
const required = value => (value == null ? 'Required' : undefined);


class AuctionForm extends Component {

  handleImages(e) {
    let image = e.target.files[0];
    const reader = new FileReader();
    reader.onload = (e) => this.props.selectImage(e.target.result, image);
    reader.readAsDataURL(e.target.files[0]);
  }

  formatData(response) {
    let fd = new FormData();
    fd.append('key', response.data.params.key);
    fd.append('file', this.props.file);
    fd.append('policy', response.data.params.policy);
    fd.append('x-amz-algorithm', response.data.params['x-amz-algorithm']);
    fd.append('x-amz-credential', response.data.params['x-amz-credential']);
    fd.append('x-amz-date', response.data.params['x-amz-date']);
    fd.append('x-amz-signature', response.data.params['x-amz-signature']);
    return fd;
  }

  onSubmit(values) {
    // Get AWS signature
    axios.get('/s3', { params: { name: this.props.file.name, type: this.props.file.type } })
      .then((response) => {
        // Format S3 data and add new image url to auction data
        const fd = this.formatData(response);
        let s3Url = `${response.data.endpoint_url}/${response.data.params.key.split(' ').join('+')}`;
        let auctionData = Object.assign({}, values, { url: s3Url });

        // Post file with signature
        axios.post(response.data.endpoint_url, fd)
          .then((awsResponse) => {
            console.log('Image(s) uploaded.');
          })
          .catch((err) => {
            alert('Error uploading image, please try again.');
          });

        this.props.deselectImage();
        // Post auction data to db
        this.props.postAuction(auctionData, () => {
          this.props.history.push('/');
        });
      });
  }

  CameraUpload(props) {
    return props.mobile ?
    <RaisedButton
      icon={<ImagePhotoCamera />}
      labelPosition="before"
      style={styles.button}
    >
      <input
        type="file"
        accept="image/*"
        capture="camera"
        style={styles.imageInput}
        onChange={props.handleImages}
      />
    </RaisedButton> :
    null;
  }

  render() {
    const { handleSubmit, reset } = this.props;

    return (
      <div style={styles.outer} >
        <div style={styles.inner}>
          <Card style={styles.card}>
            <CardMedia>
              <img src={this.props.displayImage} />
            </CardMedia>
          </Card>
          <form>
            {/*-- Photo --*/}
            <RaisedButton
              label="Choose an Image"
              labelPosition="before"
              style={styles.button}
              containerElement="label"
            >
              <input
                type="file"
                accept="image/*"
                style={styles.imageInput}
                onChange={this.handleImages.bind(this)}
              />
            </RaisedButton>
            <this.CameraUpload
              mobile={this.props.mobile}
              handleImages={this.handleImages.bind(this)}
            />
            {/*-- Category --*/}
            <div>
              <Field
                name="category"
                component={SelectField}
                hintText="Category"
                floatingLabelText="Category"
                validate={required}
                fullWidth={true}
              >
                {categories.map((category, idx) => (
                  <MenuItem key={idx} value={category[0]} primaryText={category[1]} />
                ))}
              </Field>
            </div>
            {/*-- Title --*/}
            <div>
              <Field
                name="title"
                component={TextField}
                hintText="What is it?"
                floatingLabelText="Title"
                validate={required}
                fullWidth={true}
              />
            </div>
            {/*-- Description --*/}
            <div>
              <Field
                name="description"
                component={TextField}
                hintText="What's cool about it?"
                floatingLabelText="Description"
                multiLine
                rows={2}
                validate={required}
                fullWidth={true}
              />
            </div>
            {/*-- Location --*/}
            <div>
              <Field
                name="city"
                component={TextField}
                hintText="City"
                floatingLabelText="City"
                validate={required}
                fullWidth={true}
              />
            </div>
            <div>
              <Field
                name="state"
                component={SelectField}
                hintText="State"
                floatingLabelText="State"
                validate={required}
                style={styles.state}
              >
                {states.map((state, idx) => (
                  <MenuItem key={idx} value={state} primaryText={state} />
                ))}
              </Field>
            </div>
            {/*-- End Time --*/}
            <div>
              <Field
                name="date"
                component={DatePicker}
                format={null}
                hintText="End Date"
                validate={required}
                style={styles.align}
                fullWidth={true}
              />
              {
                // <Field
                //           name="time"
                //           component={TimePicker}
                //           format={null}
                //           // Do we need this?  defaultValue={null} // TimePicker requires an object,
                //           // and redux-form defaults to ''
                //           hintText="End Time"
                //           validate={required}
                //           style={styles.align}
                //         />
                      }
            </div>
          </form>
          <div>
            <RaisedButton
              label="Submit"
              primary={true}
              onClick={handleSubmit(this.onSubmit.bind(this))}
              style={styles.button}
            />
            <Link to="/">
              <RaisedButton
                label="Cancel"
                secondary={true}
                onClick={reset}
                style={styles.button}
              />
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ images, device }) => {
  return {
    mobile: device.mobile,
    displayImage: images.displayImage,
    file: images.file
  };
};

export default reduxForm({
  form: 'PostNewAuction'
})(connect(mapStateToProps, { postAuction, selectImage, deselectImage })(AuctionForm));


