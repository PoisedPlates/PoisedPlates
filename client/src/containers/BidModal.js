import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Field, reduxForm } from 'redux-form';

import { toggleModal, fetchBid, postBid } from '../actions';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import { TextField } from 'redux-form-material-ui';

const required = value => (value == null ? 'Required' : undefined);


class BidModal extends Component {
  componentDidMount() {
    this.props.fetchBid(this.props.auction.id);
  }

  validBid(value) {
    if (value && !/^[0-9]+$/i.test(value)) {
      return 'Must be a whole number';
    } else if (value && Number(value) <= Number(this.props.currentBid)) {
      return 'Must be greater than your current bid';
    } else {
      return undefined;
    }
  }

  submitBid(input) {
    console.log('bid:', input.bid);
    this.props.postBid(this.props.auction.id, Number(input.bid));
    this.props.toggleModal();
    this.props.reset();
    this.props.fetchBid(this.props.auction.id);
  }

  handleCancel() {
    this.props.toggleModal();
    this.props.reset();
  }

  render() {
    const actions = [
      <FlatButton
        label="Cancel"
        onClick={this.handleCancel.bind(this)}
      />,
      <FlatButton
        label="Place Bid"
        primary={true}
        onClick={this.props.handleSubmit(this.submitBid.bind(this))}
      />
    ];

    return (
      <MuiThemeProvider>
        <Dialog
          title={`Ends: ${this.props.auction.end_time}`}
          actions={actions}
          modal={true}
          open={this.props.modal}
        >
          <Field
            name="bid"
            component={TextField}
            hintText="$"
            validate={[required, this.validBid.bind(this)]}
          />
        </Dialog>
      </MuiThemeProvider>
    )
  }
}

const mapStateToProps = ({ bidding }, ownProps) => {
  return {
    auction: ownProps.auction,
    modal: bidding.modal,
    currentBid: bidding.currentBid
  };
};

export { BidModal };
export default reduxForm({
  form: 'placeBid'
})(connect(mapStateToProps, { toggleModal, fetchBid, postBid })(BidModal));
