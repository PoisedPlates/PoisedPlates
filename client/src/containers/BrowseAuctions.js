import React, { Component } from 'react';
import { connect } from 'react-redux';

/* * Utils * */
import _ from 'lodash';
import { Link } from 'react-router-dom';

/* * Actions * */
import { fetchAuctions } from '../actions';

/* * Styles * */
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { GridList, GridTile } from 'material-ui/GridList';


class BrowseAuctions extends Component {

  componentDidMount() {
    this.props.fetchAuctions(this.props.showing);
  }

  renderAuctions() {
    const styles = {
      image: {
        height: '300px',
        objectFit: 'cover'
      },
      grow: {
        padding: '4px',
        width: '30rem'
      }
    }
    return _.map(this.props.auctions, auction => {
      console.log(auction)
      return (
        <div key={auction.id} style={styles.grow}>
          <Link to={`/auction/${auction.id}`}>
            <MuiThemeProvider>
              <GridTile
                key={auction.id}
                title={auction.title}
              >
                <img src={auction.url} style={styles.image} />
              </GridTile>
            </MuiThemeProvider>
          </Link>
        </div>
      )
    });
  }

  render() {
    const styles = {
      grid: {
          margin: '8px',
          display: 'flex',
          flex: 1,
          flexDirection: 'row',
          flexWrap: 'wrap',
          justifyContent: 'center'
        }
      }

    return (
      <div style={styles.grid}>
        {this.renderAuctions()}
      </div>
    )
  }
}

function mapStateToProps({ auctions, categories }) {
  return {
    auctions: auctions,
    showing: categories.showing
  };
}

export { BrowseAuctions };
export default connect(mapStateToProps, { fetchAuctions })(BrowseAuctions);




