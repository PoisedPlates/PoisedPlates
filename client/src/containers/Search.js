import React, { Component } from 'react';
import { connect } from 'react-redux';


/* * Utils * */
import { AutoComplete, Paper } from 'material-ui';

/* * Actions * */
import { queryES } from '../actions';

class Search extends Component {
    constructor(props) {
    super(props);
    this.state = {
      dataSource : [],
      inputValue : ''
    }
    this.onChange = this.onChange.bind(this);
  }

    onChange(inputValue) {
  }



  render() {
    return (
      <Paper style={ { textAlign: 'center' } } zDepth={1} rounded={false} >
        <AutoComplete
          dataSource={this.state.dataSource}
          onChange={() => console.log('change')}
          hintText="Search"
        />
      </Paper>
    )
  }
}

export default connect(null, { queryES })(Search);