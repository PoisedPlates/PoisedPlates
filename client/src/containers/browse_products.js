import _ from 'lodash';
import React, { Component } from 'react';
// import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { GridList, GridTile } from 'material-ui/GridList';


class BrowseProducts extends Component {
  constructor(props) {
    super(props);


    this.data = {
      0: {
        id: 0,
        'img-url': 'https://scontent.cdninstagram.com/t51.2885-15/e35/13767731_117068665403701_2114431511_n.jpg'
      },
      1: {
        id: 1,
        'img-url': 'https://scontent.cdninstagram.com/t51.2885-15/e35/14334643_1830784853818907_1806926123_n.jpg'
      },
      2: {
        id: 2,
        'img-url': 'https://scontent.cdninstagram.com/t51.2885-15/e35/14718580_182287928890662_5624130631819591680_n.jpg'
      },
      3: {
        id: 3,
        'img-url': 'https://scontent.cdninstagram.com/t51.2885-15/e35/14730541_209363642837720_5674879205179916288_n.jpg'
      },
      4: {
        id: 4,
        'img-url': 'https://scontent.cdninstagram.com/t51.2885-15/e35/15258783_1219743038096510_7805315801564577792_n.jpg'
      }
    };
  }

  render() {
    const styles = {
      image: {
        height: '300px',
        objectFit: 'cover'
      },
      grid: {
        margin: '8px',
        display: 'flex',
        flex: 1,
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'center',
      },
      grow: {
        padding: '4px',
        width: '30rem'
      }
    }
    return (
      <div style={styles.grid}>
        {_.map(this.data, listing => (
          <div key={listing.id} style={styles.grow}>
            <Link to={`/product/${listing.id}`}>
              <MuiThemeProvider>
                <GridTile
                  key={listing.id}
                  title="<listing.title>"
                  subtitle="listing.subtitle"
                >
                  <img src={listing['img-url']} style={styles.image}/>
                </GridTile>
              </MuiThemeProvider>
            </Link>
          </div>
        ))}
      </div>
    )
  }
}

export default BrowseProducts;