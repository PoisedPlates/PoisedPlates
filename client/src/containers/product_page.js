import React, { Component } from 'react';

export default class ProductPage extends Component {
  render() {
    return (
      <h1>Welcome to the Product Page!</h1>
    )
  }
};