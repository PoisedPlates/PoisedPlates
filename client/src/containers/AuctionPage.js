import React, { Component } from 'react';
import { connect } from 'react-redux';
import BidModal from './BidModal';

/* * Actions * */
import { fetchAuction, toggleModal } from '../actions';

/* * Styles * */
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';

class AuctionPage extends Component {
  componentDidMount() {
    if (!this.props.auction) {
      const { id } = this.props.match.params;
      this.props.fetchAuction(id);
    }
  }

  render() {
    const { auction } = this.props;
    if (!auction) {
      return <div>Loading...</div>;
    }

    return (
      <MuiThemeProvider>
        <Card>
          <CardHeader
            title={auction.title}
            // subtitle="<Rating?>"
            /*avatar="https://scontent.cdninstagram.com/t51.2885-15/e35/15258783_1219743038096510_7805315801564577792_n.jpg"*/
          />

          <CardMedia overlay={<CardTitle subtitle={`Ends: ${auction.end_time}`} />} >
            <img src={auction.images[0].url} />
          </CardMedia>

          <CardTitle title={auction.title} subtitle={`${auction.location.city}, ${auction.location.state}`} />

          <CardActions>
            <RaisedButton label="bid" secondary={true} onClick={this.props.toggleModal}/>
          </CardActions>

          <CardText>
            <BidModal auction={auction} />
            {auction.description}
          </CardText>
        </Card>
      </MuiThemeProvider>
    )
  }
};

const mapStateToProps = ({ auctions, bidding }, ownProps) => {
  return {
    auction: auctions[ownProps.match.params.id],
    bidding: bidding
  };
}

export { AuctionPage };
export default connect(mapStateToProps, { fetchAuction, toggleModal })(AuctionPage);

