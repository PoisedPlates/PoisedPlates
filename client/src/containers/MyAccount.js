import React, { Component } from 'react';
import { connect } from 'react-redux';
import BidModal from './BidModal';

/* * Actions * */
import { fetchAuction, toggleModal } from '../actions';

/* * Utils * */
import _ from 'lodash';
import { Link } from 'react-router-dom';

/* * Styles * */
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';

/* * Components * */
import AuctionStatus from '../components/AuctionStatus'

class MyAccount extends Component { //rip from AuctionPage
  componentDidMount() {
    // if (!this.props.auction) {
    //   const { id } = this.props.match.params;
    //   this.props.fetchAuction(id);
    // }
    // const post = this.props.getAuctionsByOwner
  }

  render() {
    const { auction } = this.props;
    if (!auction) {
      console.log('this >>>',this.props)
      // return <div>Loading...</div>;
    }

    return (
      <MuiThemeProvider>
        <Card>
          <CardHeader
            title={"My Accounts Page"}
          />
          {console.log('this>>',this)}
            <AuctionStatus 
              className="auctionBids" 
              type={"Auction Posts"} 
             />
            <AuctionStatus 
              className="auctionPosts" 
              type={"Auction Bids"}
            />

        </Card>
      </MuiThemeProvider>
      
    )
  }
};

const mapStateToProps = ({ auctions, bidding }, ownProps) => {
  return {
    auction: auctions[ownProps.match.params.id],
    bidding: bidding
  };
}

export { MyAccount };
export default connect(mapStateToProps, { fetchAuction, toggleModal })(MyAccount);

