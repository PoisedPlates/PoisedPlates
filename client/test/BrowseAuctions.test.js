import React from 'react';
import { BrowseAuctions } from '../src/containers/BrowseAuctions';
import renderer from 'react-test-renderer';

it('Should match its default snapshot', () => {
  const tree = renderer.create(
    <BrowseAuctions />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});