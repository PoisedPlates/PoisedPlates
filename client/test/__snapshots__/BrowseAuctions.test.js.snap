// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match its default snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "margin": "8px",
    }
  }
/>
`;
